core/templates/dev/head/pages/exploration_editor/history_tab/HistoryServices.js

```diff
@@ -205,9 +205,9 @@ oppia.factory('compareVersionsService', [
           var lca = versionsTreeService.findLCA(v1, v2);
 
           var v1States = StatesObjectFactory.createFromBackendDict(
-            v1StatesDict);
+            v1StatesDict).getStates();
```

Comment:
This one is tricky and might actually require a "real" getStates(). That said, we could call it toDict() or something like that.

The point here is that everything else just works with States and State objects, and never touches the inner raw dict. Here, toDict() then becomes an "export" method that generates a raw JS dict from the States object that can then be used by the history services. Would that work?